pkg/eventmonitor/consumers/process.go

```diff
@@ -0,0 +1,171 @@
+// instance created in cmd/system-probe/modules/eventmonitor.go:createEventMonitorModule.
+// For tests, use consumers/testutil.NewTestProcessConsumer, which also initializes the event stream for testing accordingly
+func NewProcessConsumer(id string, chanSize int, evm *eventmonitor.EventMonitor) (*ProcessConsumer, error) {
+	pc := &ProcessConsumer{
+		id:            id,
+		chanSize:      chanSize,
+		execCallbacks: callbackMap{callbacks: make(map[*ProcessCallback]struct{})},
+		exitCallbacks: callbackMap{callbacks: make(map[*ProcessCallback]struct{})},
+	}
+
+	if err := evm.AddEventConsumerHandler(pc); err != nil {
+		return nil, fmt.Errorf("cannot add event consumer handler: %w", err)
+	}
+
+	evm.RegisterEventConsumer(pc)
+
+	return pc, nil
+}
+
+// --- eventmonitor.EventConsumer interface methods
+
+// Start starts the consumer, this is a no-op for this implementation
+func (p *ProcessConsumer) Start() error {
+	return nil
+}
+
+// Stop stops the consumer, this is a no-op for this implementation
+func (p *ProcessConsumer) Stop() {
+}
+
+// ID returns the ID of the consumer
+func (p *ProcessConsumer) ID() string {
+	return p.id
+}
+
+// --- eventmonitor.EventConsumerHandler interface methods
+
+// ChanSize returns the size of the channel that the event monitor will use to send events to this handler
+func (p *ProcessConsumer) ChanSize() int {
+	return p.chanSize
+}
+
+// EventTypes returns the event types that this handler is interested in
+func (p *ProcessConsumer) EventTypes() []model.EventType {
+	return []model.EventType{model.ExecEventType, model.ExitEventType}
+}
+
+// Copy copies the event from the eventmonitor type to the one we use in this struct
+func (p *ProcessConsumer) Copy(ev *model.Event) any {
+	return &event{
+		eventType: ev.GetEventType(),
+		pid:       ev.GetProcessPid(),
+	}
+}
+
+// HandleEvent handles the event from the event monitor
+func (p *ProcessConsumer) HandleEvent(ev any) {
+	sevent, ok := ev.(*event)
+	if !ok {
+		return
+	}
+
+	switch sevent.eventType {
+	case model.ExecEventType:
+		p.execCallbacks.call(sevent.pid)
+	case model.ExitEventType:
+		p.exitCallbacks.call(sevent.pid)
+	}
+}
+
+// SubscribeExec subscribes to process exec events, and returns the function
+// that needs to be called to unsubscribe
+func (p *ProcessConsumer) SubscribeExec(callback ProcessCallback) func() {
+	return p.execCallbacks.add(callback)
+}
+
+// SubscribeExit subscribes to process exit events, and returns the function
+// that needs to be called to unsubscribe
+func (p *ProcessConsumer) SubscribeExit(callback ProcessCallback) func() {
+	return p.exitCallbacks.add(callback)
+}
+
+// callbackMap is a helper struct that holds a map of callbacks and a mutex to protect it
+type callbackMap struct {
+	// callbacks holds the set of callbacks
+	callbacks map[*ProcessCallback]struct{}
+
+	// mutex is the mutex that protects the callbacks map
+	mutex sync.RWMutex
```

Comment:
what is the common pattern to use this struct?
in some cases rwmutex perform worse than the regular mutex

Reply:
I took the same pattern as there is in process-monitor, I assumed it was tested to be the best option.

Reply:
cc @guyarb @vitkyrka

Reply:
common use case - adding all callbacks (write lock), and then calling them (read lock)

normally, all callbacks are being added (or removed) before (or after) the operation.
We can use regular mutex here, but, if we do intend to make the change - it should be tested to show it improves the usage of rwmutex

Reply:
me and @brycekahle  discussed rwmutex in some other context. and in some cases the rwmutex actually creates slight overhead (even when just using the readerlock without writer lock).

Reply:
>  and in some cases the rwmutex actually creates slight overhead (even when just using the readerlock without writer lock).

What cases? is it relevant here?

Reply:
my previous comment wasn't clear. we might want to avoid using the rwmutex in the callback struct at all and instead consider non thread safe design, leaving it to the caller (we might have to change the current implementation a bit, as the struct is encapsulated)  

about rwmutex [reference](https://zephyrtronium.github.io/articles/rwmutex.html):

Reply:
I don't think we can do that, we're always going to (potentially) have two goroutines accessing the callback map concurrently, one from the event stream and another one from the subscription.

Reply:
do we have any subscriptions happening in runtime? or rather we can limit the subscription during the "init" phase before the handler go-routine starts?

Reply:
also [these](https://github.com/ntsd/go-mutex-comparison?tab=readme-ov-file#test-scenarios) benchmarks for reference

Reply:
It might not always be possible to limit the subscriptions, specially considering that the current way of adding event stream consumers requires having a global variable to share the consumer, and that could probably change later.  Also, it'd require changing the `SubscribeX` signatures to be able to return an error.  

I think it's a good idea to research this a bit more, but I think we should have a better view of dependencies in system-probe modules first to see how that fits with those phases.

Reply:
i agree that we can leave it out of scope for this PR. Resolving the comment